Recast/HomeViewController.swift

```diff
@@ -0,0 +1,219 @@
+    var recommendedPodcasts: [DummyPodcastSeries] = [] {
+        didSet {
+            guard let cv = view.viewWithTag(102) as? UICollectionView else { return }
+            cv.reloadData()
+        }
+    }
+
+    // Table View reuse identifiers
+    let continueTvReuse = "continueListeningTvReuse"
+    let gridTvReuse = "gridTvReuse"
+
+    // Collection View reuse identifiers
+    let continueCvReuse = "continueListeningCvReuse"
+    let gridCvReuse = "gridCvReuse"
+
+    // MARK: - Lifecycle
+//    override var preferredStatusBarStyle: UIStatusBarStyle {
+//        return .lightContent
+//    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // VC settings
+        title = "Home"
+        view.backgroundColor = .black
+        navigationController?.navigationBar.prefersLargeTitles = true
+
+        // homeTableView
+        homeTableView = UITableView(frame: .zero, style: .grouped)
+        homeTableView.delegate = self
+        homeTableView.dataSource = self
+        homeTableView.allowsSelection = false
+        homeTableView.backgroundColor = .black
+        homeTableView.separatorColor = .black
+        homeTableView.register(ContinueTableViewCell.self, forCellReuseIdentifier: continueTvReuse)
+        homeTableView.register(SeriesGridTableViewCell.self, forCellReuseIdentifier: gridTvReuse)
+        view.addSubview(homeTableView)
+
+        prepareDummy()
+        setUpConstraints()
+    }
+
+    func setUpConstraints() {
+        homeTableView.snp.makeConstraints { (make) -> Void in
+            make.edges.equalToSuperview()
+        }
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+
+    func prepareDummy() {
+        continuePodcasts.append(dummyContinue1)
+        continuePodcasts.append(dummyContinue2)
+        favoritePodcasts.append(dummyContinue1)
+        favoritePodcasts.append(dummyContinue2)
+        recommendedPodcasts.append(dummyContinue1)
+        recommendedPodcasts.append(dummyContinue2)
+    }
+}
+
+// MARK: - TV Delegate
+extension HomeViewController: UITableViewDelegate {
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 3
+    }
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let headerView = HomeTableViewHeaderView()
+        if let sectionType = headerTypes[section] {
+            headerView.configure(for: sectionType)
+        }
+        return headerView
+    }
+
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return 60
+    }
+
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return 108
+    }
+
+    func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        switch indexPath.section {
+        case 0:
```

Comment:
i think we want to avoid switching on the raw section, see this for formatting tableviews: https://bible.cornellappdev.com/ios/tableviews

Reply:
^^